src/Auth.ts

```diff
@@ -339,206 +194,430 @@ export default class Auth {
+        if (typeof this.clientStorage.setItemAsync !== "function") {
+          console.error(`this.clientStorage.setItemAsync is not a function`);
+          break;
+        }
+        this.clientStorage.setItemAsync(key, value);
+        break;
+      default:
+        break;
+    }
+  }
+
+  private async _getItem(key: string): Promise<unknown> {
+    switch (this.clientStorageType) {
+      case "web":
+        if (typeof this.clientStorage.getItem !== "function") {
+          console.error(`this.clientStorage.getItem is not a function`);
+          break;
+        }
+        return this.clientStorage.getItem(key);
+      case "react-native":
+        if (typeof this.clientStorage.getItem !== "function") {
+          console.error(`this.clientStorage.getItem is not a function`);
+          break;
+        }
+        return await this.clientStorage.getItem(key);
+      case "capacitor":
+        if (typeof this.clientStorage.get !== "function") {
+          console.error(`this.clientStorage.get is not a function`);
+          break;
+        }
+        const res = await this.clientStorage.get({ key });
+        return res.value;
+      case "expo-secure-storage":
+        if (typeof this.clientStorage.getItemAsync !== "function") {
+          console.error(`this.clientStorage.getItemAsync is not a function`);
+          break;
+        }
+        return this.clientStorage.getItemAsync(key);
+      default:
+        break;
+    }
+  }
+
+  private async _removeItem(key: string): Promise<void> {
+    switch (this.clientStorageType) {
+      case "web":
+        if (typeof this.clientStorage.removeItem !== "function") {
+          console.error(`this.clientStorage.removeItem is not a function`);
+          break;
+        }
+        return this.clientStorage.removeItem(key);
+      case "react-native":
+        if (typeof this.clientStorage.removeItem !== "function") {
+          console.error(`this.clientStorage.removeItem is not a function`);
+          break;
+        }
+        return await this.clientStorage.removeItem(key);
+      case "capacitor":
+        if (typeof this.clientStorage.remove !== "function") {
+          console.error(`this.clientStorage.remove is not a function`);
+          break;
+        }
+        await this.clientStorage.remove({ key });
+        break;
+      case "expo-secure-storage":
+        if (typeof this.clientStorage.deleteItemAsync !== "function") {
+          console.error(`this.clientStorage.deleteItemAsync is not a function`);
+          break;
+        }
+        this.clientStorage.deleteItemAsync(key);
+        break;
+      default:
+        break;
+    }
+  }
+
+  private _generateHeaders(): null | types.Headers {
+    if (this.useCookies) return null;
+
+    return {
+      Authorization: `Bearer ${this.currentSession.getSession()?.jwt_token}`,
+    };
+  }
+
+  private _autoLogin(refreshToken: string | null): void {
+    if (this.ssr) {
+      this._clearSession();
```

Comment:
Skip this one on SSR. Will only show an error that it' can't clear the session.

Reply:
Just to be sure, you want to remove `this._clearSession();` but keep the `return` when `this.ssr`?

Reply:
Yes

Reply:
👍